cmgd/cmgd_vty.c

```diff
@@ -307,19 +307,57 @@ DEFPY(cmgd_commit_abort,
 }
 
 DEFPY(cmgd_set_config_data,
-	  cmgd_set_config_data_cmd,
-	  "cmgd set-config xpath WORD$path value WORD$val",
-	  CMGD_STR
-	  "Set configuration data\n"
-	  "XPath expression specifying the YANG data path\n"
-	  "XPath string\n"
-	  "Value of the data to set to\n"
-	  "<value of the data>\n")
+      cmgd_set_config_data_cmd,
+      "cmgd set-config xpath WORD$path value WORD$val",
+      CMGD_STR
+      "Set configuration data\n"
+      "XPath expression specifying the YANG data path\n"
+      "XPath string\n"
+      "Value of the data to set to\n"
+      "<value of the data>\n")
 {
-	strlcpy(vty->cfg_changes[0].xpath, path,
-		sizeof(vty->cfg_changes[0].xpath));
-	vty->cfg_changes[0].value = val;
-	vty->num_cfg_changes = 1;
+	struct nb_cfg_change *change;
+
+	if (vty->num_cfg_changes == VTY_MAXCFGCHANGES) {
```

Comment:
Why is this check required? vty->cum_cfg_changes should always be overwritten with 1.